extensions/standard-processors/modbus/ReadModbusFunctions.h

```diff
@@ -69,7 +69,7 @@ class ReadModbusFunction {
   const uint8_t unit_id_;
 };
 
-class ReadCoilStatus final : public ReadModbusFunction {
+class ReadCoilStatus : public ReadModbusFunction {
```

Comment:
Due to a bug in the latest apple clang final classes break dynamic_cast 
https://stackoverflow.com/questions/79192304/macos-xcode-16-breaks-dynamic-cast-for-final-types-defined-in-shared-library
Should we `-fno-assume-unique-vtables` or just remove the finals here (we rely on dynamic_cast for tests only) but this might also come up somewhere that is not covered by tests.

Reply:
I think removing `final` in the affected cases is fine, and better if there are only few cases. `-f` flags change the compiler in incompatible ways, it's best to avoid them IMO.